app/components/TumourSummaryEdit/index.tsx

```diff
@@ -150,6 +148,7 @@ const TumourSummaryEdit = ({
         if (mutationBurden?.ident) {
           apiCalls.push(api.put(`/reports/${report.ident}/mutation-burden/${mutationBurden.ident}`, newMutationBurdenData, {}));
         } else {
+          newMutationBurdenData.role = 'primary'; // Default role of new mutation burden data for reports with no existing analysis per ClinInfo team
```

Comment:
This works, but that's a nono in React, if you need to change an attribute of an object in a state, you should use destructuring -- but even in this case 
`newMutationBurdenData.role = { ...newMutationBurdenData, role: 'primary' }` is unacceptable because it violates React principles where you should use the set<x> function (setNewMutationBurden) to change the state

instead of 
```
apiCalls.push(api.post(`/reports/${report.ident}/mutation-burden`, newMutationBurdenData, {}));
```

do
```
apiCalls.push(api.post(`/reports/${report.ident}/mutation-burden`, { ...newMutationBurdenData, role: 'primary' }, {}));
```

Reply:
This is the elegant solution I was looking for. I'll make the proposed change now, thanks Jacky!